src/main/java/s3NativeClient/com/amazonaws/s3/S3NativeClient.java

```diff
@@ -199,12 +200,15 @@ public int onResponseBody(ByteBuffer bodyBytesIn, long objectRangeStart, long ob
             }
 
             @Override
-            public void onFinished(int errorCode, int responseStatus, byte[] errorPayload) {
-                CrtS3RuntimeException ex = null;
+            public void onFinished(S3FinishedResponseContext context) {
+                CrtRuntimeException ex = null;
                 try {
-                    if (errorCode != CRT.AWS_CRT_SUCCESS) {
-                        ex = new CrtS3RuntimeException(errorCode, responseStatus, errorPayload);
+                    if (context.getResponseStatus() > 299){
```

Comment:
Why 300+? 3xx are redirection codes and are generally not considered errors. not sure if we get any 3xx from s3.